In [ ]:
!pip install brainiak

In [1]:
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
#from brainiak.funcalign.srm import SRM
#from brainiak.isc import isc

In [10]:
roi = 'A1'
filepath = 'drive/MyDrive/fMRI_music_data/%s_by_subject_sorted_conds/'%roi
subjects = ['03','15','20','23']
n_sub = len(subjects)

In [7]:
# load data from one subject for a quick sanity check
this_subj = subjects[0]
test = np.load(filepath+'%s_sub-1%s.npy'%(roi,this_subj))
print(test.shape)

(516, 1776)


There are 516 voxels in A1. There are 1776 total time points: 148 TRs per run * 12 runs.

In [46]:
# load in the data from all subjects
data = []
for subj in subjects: 
  data.append(np.load(filepath+'%s_sub-1%s.npy'%(roi,this_subj)))

# recast the list into an array and check the shape
data = np.asarray(data)
print(data.shape)

(4, 516, 1776)


Reshape the data so that it's subjects x voxels x TRs x conds x runs. This gives flexibility to select training and testing data for SRM later.

In [47]:
n_voxels = data.shape[1]
n_TRs = 148
n_conds = 4
n_runs = 3

orig_data = np.copy(data)
orig_data = np.reshape(orig_data,(n_sub,n_voxels,n_TRs,n_conds,n_runs),order='F')
print(orig_data.shape)

(4, 516, 148, 4, 3)


As long as this array acts like the toy example (see earlier version), then `order = 'F'` should keep the individual runs intact during the reshaping.

In [35]:
toy = np.arange(60)
toy_re = np.reshape(toy,(5,4,3),order='F')
print(toy_re)

[[[ 0 20 40]
  [ 5 25 45]
  [10 30 50]
  [15 35 55]]

 [[ 1 21 41]
  [ 6 26 46]
  [11 31 51]
  [16 36 56]]

 [[ 2 22 42]
  [ 7 27 47]
  [12 32 52]
  [17 37 57]]

 [[ 3 23 43]
  [ 8 28 48]
  [13 33 53]
  [18 38 58]]

 [[ 4 24 44]
  [ 9 29 49]
  [14 34 54]
  [19 39 59]]]


In [37]:
toy_rere = np.reshape(toy_re,-1,order='F')
print(toy_rere)

[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47
 48 49 50 51 52 53 54 55 56 57 58 59]


In [39]:
toy1 = np.reshape(toy,(20,3),order='F')
print(toy1)

[[ 0 20 40]
 [ 1 21 41]
 [ 2 22 42]
 [ 3 23 43]
 [ 4 24 44]
 [ 5 25 45]
 [ 6 26 46]
 [ 7 27 47]
 [ 8 28 48]
 [ 9 29 49]
 [10 30 50]
 [11 31 51]
 [12 32 52]
 [13 33 53]
 [14 34 54]
 [15 35 55]
 [16 36 56]
 [17 37 57]
 [18 38 58]
 [19 39 59]]


In [41]:
print(toy1[:,0])

[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19]


In [45]:
toy2 = np.reshape(toy1,(5,4,3),order='F')
print(toy2[:,:,0])

[[ 0  5 10 15]
 [ 1  6 11 16]
 [ 2  7 12 17]
 [ 3  8 13 18]
 [ 4  9 14 19]]
